### PR TITLE
Adds CreateDeployment Operation

### PR DESCRIPTION
**Pull request by rain-on**
Includes the DeploymentResource, DeploymentResourceWithLinks and
DeploymentsPaginatedCollection and DeploymentsApi.
    
This allows users of the SDK to call into the ExtensionsCreateApi
in order to create a deployment.

### CODE REVIEW

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandModel.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import com.google.gson.annotations.SerializedName;
+
+@SuppressWarnings("UnusedVariable")
+public class BaseCommandModel {
+
+  @SerializedName("projectNameOrId")
+  private String projectName;
+
+  @SerializedName("environmentIdsOrNames")
+  private List<String> environmentName;
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandModel.java:32]: Pluralise

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandModel.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.gson.annotations.SerializedName;
+
+@SuppressWarnings("UnusedVariable")
+public class BaseCommandModel {
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandModel.java:26]: Fields a mix of `final` and not `final`

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandModel.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  private String projectName;
+
+  @SerializedName("environmentIdsOrNames")
+  private List<String> environmentName;
+
+  @SerializedName("tenantIdsOrNames")
+  private Optional<String> tenantName;
+
+  @SerializedName("tenantTags")
+  private List<String> tenantTags;
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandModel.java:38]: Optional?

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.gson.annotations.SerializedName;
+
+@SuppressWarnings("UnusedVariable")
+public class BaseCommandParameters {
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java:26]: mixed usage of `final`?

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+  @SerializedName("projectNameOrId")
+  private String projectName;
+
+  @SerializedName("environmentIdsOrNames")
+  private List<String> environmentName;
+
+  @SerializedName("tenantIdsOrNames")
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java:34]: Should this be a `List<>`

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParameters.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.gson.annotations.SerializedName;
+
+@SuppressWarnings("UnusedVariable")
+public class CreateDeploymentCommandParameters extends BaseCommandParameters {
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParameters.java:26]: Outstanding question about whether the SDK needs to handle IdOrName as opposed to Name only

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParametersBuilder.java [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+package com.octopus.sdk.model.commands;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class CreateDeploymentCommandParametersBuilder {
+  private String projectName;
+  private List<String> environmentName;
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParametersBuilder.java:25]: Pluralise

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParametersBuilder.java [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+  private Optional<String> tenantName;
+  private List<String> tentantTags;
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParametersBuilder.java:27]: See BaseCommandParameters, should this be a list and pluraised

**Review by long-sean — COMMENTED** [review]
Some comments that need addressing in terms of what is a collection etc.

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java (hunk 4 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java:26]: ditched final all together on this one

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java (hunk 5 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/model/commands/BaseCommandParameters.java:34]: Is now

File: octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParameters.java (hunk 6 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/model/commands/CreateDeploymentCommandParameters.java:24]: updated to be NameOrId (and I hate it)

File: acceptance-tests/src/test/java/com/octopus/sdk/test/CreateDeploymentAcceptanceTest.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+      final ReleaseApi releaseApi = ReleaseApi.create(client, spaceHome);
+      final ReleaseResourceWithLinks release = new ReleaseResourceWithLinks();
+      release.setVersion(releaseVersion);
+      release.setProjectId(projectCreated.getId());
+
+      releaseApi.create(release);
+
+    } catch (final Exception e) {
+      LOG.error(e);
+      deleteSpaceValidly(spacesOverviewApi, toCreate);
```
Comment (long-sean) [inline acceptance-tests/src/test/java/com/octopus/sdk/test/CreateDeploymentAcceptanceTest.java:85]: `toCreate` never changes from `null` Perhaps call method with `null` directly

**Review by long-sean — COMMENTED** [review]
New comment on CreateDeploymentAcceptanceTest, otherwise all good

File: acceptance-tests/src/test/java/com/octopus/sdk/test/CreateDeploymentAcceptanceTest.java (hunk 9 above)
Comment (rain-on) [reply acceptance-tests/src/test/java/com/octopus/sdk/test/CreateDeploymentAcceptanceTest.java:85]: this is now in the SpaceScoped file :/

have updated to better handle deleting a created space

**Review by long-sean — APPROVED** [review]

**Pull request merged**